src/components/_classes/component/Component.js

```diff
@@ -1194,22 +1202,27 @@ export default class Component extends Element {
     }
   }
 
+  createTooltip(tooltipEl,  settings = {}) {
+    const tooltipAttribute = tooltipEl.getAttribute('data-tooltip');
+    const tooltipDataTitle = tooltipEl.getAttribute('data-title');
+    const tooltipText = this.interpolate(tooltipDataTitle || tooltipAttribute)
+                            .replace(/(?:\r\n|\r|\n)/g, '<br />');
+
+    return tippy(tooltipEl, {
+      allowHTML: true,
+      trigger: 'mouseenter click focus',
+      placement: 'right',
+      zIndex: 10000,
+      interactive: true,
+      ...settings,
+      content: this.t(this.sanitize(tooltipText), { _userInput: true }),
+    });
+  }
+
   attachTooltips(toolTipsRefs) {
     toolTipsRefs?.forEach((tooltip, index) => {
```

Comment:
I like that you factored out the `createToolTip` function; it makes it much cleaner.  You can go step further if you want by chaining some array methods together.
```suggestion
    this.tooltips = toolTipsRefs.filter(Boolean).map(this.createTooltip)
```
You might need to `bind` or `call` the `createTooltip`. I can never remember for objects.

Just a suggestion, feel free to ignore

Reply:
Refactored with lodash.